app/src/main/java/org/simple/clinic/activity/RxActivityLifecycle.kt

```diff
@@ -62,3 +31,42 @@ class RxActivityLifecycle internal constructor(private val events: Observable<Ac
     }
   }
 }
+
+private class RxActivityLifecycleCallbacks(
+    private val theActivity: AppCompatActivity,
+    private val emitter: ObservableEmitter<ActivityLifecycle>
+) : SimpleActivityLifecycleCallbacks() {
+
+  private val activityName: String = theActivity.javaClass.simpleName
+
+  override fun onActivityResumed(activity: Activity) {
+    if (activity === theActivity) {
```

Comment:
Do we still need this check? Wouldn't it be possible to listen to a specific activity events by using a filter on the lifecycle stream?

Reply:
Discussed offline.